metrics/internal/cache/pv.go

```diff
@@ -94,8 +94,14 @@ func appendIfNotExists(slice []string, value string) []string {
 
 // Add inserts to the PersistentVolumeStore.
 func (p *PersistentVolumeStore) Add(obj interface{}) error {
-	pv, ok := obj.(*corev1.PersistentVolume)
-	if !ok {
+	var pv *corev1.PersistentVolume
+	// obj can be of PV type or a pointer to it
+	switch pvFromObj := obj.(type) {
+	case *corev1.PersistentVolume:
+		pv = pvFromObj
+	case corev1.PersistentVolume:
+		pv = &pvFromObj
+	default:
```

Comment:
We never needed this before. Why do we need it now?
What change caused us to require this?

Reply:
We have a long standing BZ raised for this : https://bugzilla.redhat.com/show_bug.cgi?id=2208302
It seems like issue is there for some time